crates/consensus/src/engine/mod.rs

```diff
@@ -194,9 +202,29 @@ impl<Client: HeaderProvider + BlockProvider> ConsensusEngine for EthConsensusEng
             })
         }
 
-        // TODO: execute block
-
-        Ok(PayloadStatus::new(PayloadStatusEnum::Valid, block.hash()))
+        let (header, body, _) = block.split();
+        let transactions = body
+            .into_iter()
+            .map(|tx| {
+                let tx_hash = tx.hash;
+                tx.into_ecrecovered().ok_or(EngineApiError::PayloadSignerRecovery { hash: tx_hash })
+            })
+            .collect::<Result<Vec<_>, EngineApiError>>()?;
+        let state_provider = SubState::new(State::new(&*self.client));
```

Comment:
I don't love exposing the revm internals like Substrate here or in the executor. Would love a way to have a less leaky abstraction to get a state provider, or rename it to something more descriptive.

Reply:
left comment in a relevant issue https://github.com/paradigmxyz/reth/issues/627#issuecomment-1367161635